runtime/src/commonMain/kotlin/pbandk/internal/binary/kotlin/KotlinBinaryWireDecoder.kt

```diff
@@ -254,3 +246,20 @@ internal class KotlinBinaryWireDecoder(private val wireReader: WireReader) : Bin
         }
     }
 }
+
+public data class ReturnRawByte(val b: Byte, val eof: Boolean)
+
+public fun readerRawVarint64(readRawByte: () -> ReturnRawByte): Long {
```

Comment:
Since varint decoding is in the critical path for lots of protobuf operations, it's worth marking this function `inline` to avoid the runtime overhead of the lambda parameter (see the intro of https://kotlinlang.org/docs/inline-functions.html for more details on the overhead).

Reply:
Adding inline then gives error message on throw InvalidProtocolBufferException.malformedVarint() :

Public-API inline function cannot access non-public-API 'internal constructor InvalidProtocolBufferException(message: String) defined in pbandk.InvalidProtocolBufferException'

because InvalidProtocolBufferException.malformedVarint()  is not public. Do you want to make one of the constructors public, or switch to a RuntimeException, or ?